### PR TITLE
Dto enhancement

### PR DESCRIPTION
**Pull request by supragyan**
1.Created "StringFromPrimitivetype" and "PrimitivetypeFromString" method.
2.Reduced duplicated Code that convert to a string from primitive type.
3.Reduced duplicated Code that convert to a primitive type from string.
4.Created method to support JSon data in JQGrid.
5.Created sample code in ProjectX_sample/Aspx/testFxLayerP/table location.

### CODE REVIEW

File: root/files/resource/Sql/sqlserver/ScreenAutoGen/CreateTable.sql [hunk 1]
```diff
@@ -0,0 +1,24 @@
+/*********************************************************************************/
```
Comment (yusukemaegawa) [inline root/files/resource/Sql/sqlserver/ScreenAutoGen/CreateTable.sql]: Why is this SQL necessary for DTO enhancement?

File: root/files/resource/Sql/sqlserver/ScreenAutoGen_FreeSQL/CreateTable.sql [hunk 2]
```diff
@@ -0,0 +1,66 @@
+/*********************************************************************************/
```
Comment (yusukemaegawa) [inline root/files/resource/Sql/sqlserver/ScreenAutoGen_FreeSQL/CreateTable.sql]: Why is this SQL necessary for DTO enhancement?

File: root/files/resource/Sql/sqlserver/ScreenAutoGen_FreeSQL/DaoTABL2_TABL1_JOIN_S2_Select.xml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (yusukemaegawa) [inline root/files/resource/Sql/sqlserver/ScreenAutoGen_FreeSQL/DaoTABL2_TABL1_JOIN_S2_Select.xml]: Why is this SQL necessary for DTO enhancement?

File: root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTable.cs [hunk 4]
```diff
@@ -400,5 +407,154 @@ public void AcceptChanges()
                 }
             }
         }
+
+        #region GetJson
+
+        /// <summary>
+        /// GetJson method for selecting data
```
Comment (yusukemaegawa) [inline root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTable.cs]: I was not able to understand the purpose of this method. (what does this method do?) Please comment on the purpose of this method. Please think about modifying method name to the "understandable" name.

File: root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTable.cs [hunk 5]
```diff
@@ -400,5 +407,154 @@ public void AcceptChanges()
...
+        /// </summary>
+        /// <returns></returns>
+        public JQGridDataClass GetJson()
+        {
+            // サンプルデータ取得
+            System.Data.DataTable dt = this.CreateSampleData();
```
Comment (yusukemaegawa) [inline root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTable.cs]: Please do not use the sample data. Please use the data which you acquired using TableAdapter.

File: root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTable.cs [hunk 6]
```diff
@@ -400,5 +407,154 @@ public void AcceptChanges()
...
+
+        #endregion
+
+        #region CreateSampleData
+
+        /// <summary>
+        /// Creates Sample Data in datatable.
+        /// </summary>
+        /// <returns></returns>
+        private System.Data.DataTable CreateSampleData()
```
Comment (yusukemaegawa) [inline root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTable.cs]: Please do not use the sample data. Please delete this method.

File: root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTable.cs [hunk 7]
```diff
@@ -400,5 +407,154 @@ public void AcceptChanges()
...
+        {
+            /// <summary>
+            /// 列の固有 ID
+            /// </summary>
+            public string id;
+
+            /// <summary>
+            /// 列のデータを含むリスト
+            /// </summary>
+            public System.Collections.Generic.List<string> cell;
```
Comment (yusukemaegawa) [inline root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTable.cs]: Please revise data type to support the data type except the string type.

File: root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTable.cs [hunk 8]
```diff
@@ -400,5 +407,154 @@ public void AcceptChanges()
...
+        /// GetJson method for selecting data
+        /// </summary>
+        /// <returns></returns>
+        public JQGridDataClass GetJson()
+        {
+            // サンプルデータ取得
+            System.Data.DataTable dt = this.CreateSampleData();
+
+            // クエリ文字列から、グリッドの表示に関する情報を取得
+            System.Collections.Specialized.NameValueCollection queryStrings = HttpContext.Current.Request.QueryString;
```
Comment (yusukemaegawa) [inline root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTable.cs]: Does this method support a search (search criteria) of jqGrid?

File: root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTables.cs [hunk 9]
```diff
@@ -65,7 +73,7 @@ public class TableRecords
         #region インスタンス変数
 
         /// <summary>表を保持するList</summary>
-        private List<DTTable> _tbls = new List<DTTable>();
+        public List<DTTable> _tbls = new List<DTTable>();
```
Comment (yusukemaegawa) [inline root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTables.cs]: I can acquire the table included in DTTables by indexer.
Why will it be necessary to declare this variable with "Public"?

File: root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTables.cs [hunk 10]
```diff
@@ -157,6 +167,7 @@ public IEnumerator GetEnumerator()
         public string SaveJson()
```
Comment (yusukemaegawa) [inline root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTables.cs]: Please add comment to understand contents of the processing.

File: root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTable.cs (hunk 4 above)
Comment (daisukenishino) [inline root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTable.cs]: "GetJson" method is incorrect name,
"SavejqGridJson" method is correct name.

Also about the "SavejqGridJson" 's contents, please refer to the following URL.
https://github.com/OpenTouryoProject/OpenTouryo/issues/68#issuecomment-73005844

File: root/programs/C#/Samples/ProjectX_sample/Aspx/testFxLayerP/table/JQGridHandler.ashx [hunk 11]
```diff
@@ -0,0 +1,72 @@
+﻿<%@ WebHandler Language="C#" Class="JQGridHandler" %>
```
Comment (daisukenishino) [inline root/programs/C#/Samples/ProjectX_sample/Aspx/testFxLayerP/table/JQGridHandler.ashx]: I do not want to use the "*.ashx" of HttpHandler.
That you use the "*.svc.cs" of WCF.

File: root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTable.cs (hunk 4 above)
Comment (daisukenishino) [inline root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTable.cs]: - Do not do data access in GetJson (SavejqGridJson) method.
- Data access is performed on the module of WebAPI(".svc.cs" of WCF).

File: root/programs/C#/Samples/ProjectX_sample/Aspx/testFxLayerP/table/JQGridHandler.ashx (hunk 11 above)
Comment (daisukenishino) [inline root/programs/C#/Samples/ProjectX_sample/Aspx/testFxLayerP/table/JQGridHandler.ashx]: We found ways that WCF's output string are not granted "double quotes".
Please refer to the following URL.
https://github.com/OpenTouryoProject/OpenTouryo/issues/68#issuecomment-71993115

File: root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTable.cs (hunk 4 above)
Comment (supragyan) [inline root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTable.cs]: Do we need to data access in new WebAPI module or existing project 'SampleProgram\ASPNET\WebForm\jqGridandWCF\jqGridandWCF\JSONService.svc.cs.

Please confirm.

Comment (daisukenishino) [inline root/programs/C#/Frameworks/Infrastructure/Public/Dto/DTTable.cs]: - You need to create a WebAPI module of WCF.  
/root/programs/C#/Samples/WebApp_sample/ProjectX_sample\WebService\JSONService.svc (.cs file).
- And, you get the Datatable perform data access in this module.

File: root/programs/C#/Frameworks/Infrastructure/Public/Util/CustomMarshaler.cs [hunk 12]
```diff
@@ -223,5 +229,107 @@ public static byte[] StructuresToBytes(object[] cStructures, int count)
...
+                    convertedString = Convert.ToBase64String((byte[])primitiveType);
+                }
+                else if (DTColumn.CheckType(primitiveType, DTType.String))
+                {
+                    if (checkType == true)
+                    {
+                        convertedString = primitiveType.ToString().Replace("\r", "\rrnr:");
+                        convertedString = primitiveType.ToString().Replace("\n", "\rrnn:");
+
+                        convertedString = primitiveType.ToString().Replace("\r", "\r\n");
```
Comment (yusukemaegawa) [inline root/programs/C#/Frameworks/Infrastructure/Public/Util/CustomMarshaler.cs]: You substitute the value that you replaced from "primitiveType" for "convertedString" three times.
Therefore, the 266th line and the 267th line source code do not have a meaning.
We think that this is a bug.

File: root/programs/C#/Frameworks/Infrastructure/Public/Util/CustomMarshaler.cs [hunk 13]
```diff
@@ -223,5 +229,107 @@ public static byte[] StructuresToBytes(object[] cStructures, int count)
...
+                {
+                    DateTime dateTime = (DateTime)primitiveType;
+                    convertedString += dateTime.Year + "/";
+                    convertedString += dateTime.Month + "/";
+                    convertedString += dateTime.Day + "-";
+
+                    convertedString += dateTime.Hour + ":";
+                    convertedString += dateTime.Minute + ":";
+                    convertedString += dateTime.Second + ".";
+                    convertedString += dateTime.Millisecond;
```
Comment (yusukemaegawa) [inline root/programs/C#/Frameworks/Infrastructure/Public/Util/CustomMarshaler.cs]: You can write the same processing as this in one line when you use DateTime.ToString(format) method.
Therefore, we recommend that you use DateTime.ToString method.

File: root/programs/C#/Frameworks/Infrastructure/Public/Util/CustomMarshaler.cs [hunk 14]
```diff
@@ -223,5 +229,107 @@ public static byte[] StructuresToBytes(object[] cStructures, int count)
...
+                    string hmsf = cellString.Split('-')[1];
+
+                    DateTime cellDttm = new DateTime(
+                        int.Parse(ymd.Split('/')[0]),
+                        int.Parse(ymd.Split('/')[1]),
+                        int.Parse(ymd.Split('/')[2]),
+                        int.Parse(hmsf.Split(':')[0]),
+                        int.Parse(hmsf.Split(':')[1]),
+                        int.Parse(hmsf.Split(':')[2].Split('.')[0]),
+                        int.Parse(hmsf.Split(':')[2].Split('.')[1]));
```
Comment (yusukemaegawa) [inline root/programs/C#/Frameworks/Infrastructure/Public/Util/CustomMarshaler.cs]: You can write the same processing as this in one line when you use DateTime.Parse(format) method.
Therefore, we recommend that you use DateTime.Parse method.

File: root/programs/C#/Samples/ProjectX_sample/Aspx/Common/testBlankScreen.master [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+    <script type="text/javascript" src="/ProjectX_sample/Framework/js/ie_key_event.js"></script>
+    <!-- IE8をIE7互換モードで動作させる -->
+    <!--meta http-equiv="X-UA-Compatible" content="IE=EmulateIE7" /-->
+    <link rel="stylesheet" href="/ProjectX_sample/Css/style.css" type="text/css" />
+    <!--FxCode：add-end-->
+    <script src="../../../JQGridScripts/grid.locale-en.js" type="text/javascript"></script>
+    <script src="../../../JQGridScripts/jquery-1.9.0.min.js" type="text/javascript"></script>
+    <link href="../../JQGridScripts/jquery-ui-1.9.2.custom.css" rel="stylesheet" type="text/css" />
+    <script src="../../../JQGridScripts/jquery.jqGrid.js" type="text/javascript"></script>
+    <link href="../../JQGridScripts/ui.jqgrid.css" rel="stylesheet" type="text/css" />
```
Comment (yusukemaegawa) [inline root/programs/C#/Samples/ProjectX_sample/Aspx/Common/testBlankScreen.master]: You refer to a JavaScript and CSS of jqGrid in a master page.
Do you want to define it for "a master page for jqGrid"?
If so, please make a master page name understandable.

File: root/programs/C#/Samples/ProjectX_sample/Aspx/testFxLayerP/table/testJQGridJson.aspx [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+            colModel: [
+                    { name: 'id' },
+                    { name: 'name', sortable: false },
+                    { name: 'email', sortable: false },
+                    { name: 'phone', sortable: false },
+
+            ],
+            rowNum: 10,
+            mtype: 'GET',
+            loadonce: true,
```
Comment (yusukemaegawa) [inline root/programs/C#/Samples/ProjectX_sample/Aspx/testFxLayerP/table/testJQGridJson.aspx]: Please set "loadonce" to "false". (paging does not work well when "loadonce" is set to "true")

**Pull request closed without merging**